contributors/design-proposals/daemonset-update.md

```diff
@@ -270,22 +292,40 @@ Another way to implement DaemonSet history is through creating `PodTemplates` as
 snapshots of DaemonSet templates, and then create them in DaemonSet controller:
 
 - Find existing PodTemplates whose labels are matched by DaemonSet
-  `.spec.selector`
+  `.spec.selector`.
   - Sort those PodTemplates by creation timestamp and only retain at most
-   `.spec.revisionHistoryLimit` latest PodTemplates (remove the rest)
-  - Find the PodTemplate whose `.template` is the same as DaemonSet
-   `.spec.template`. If not found, create a new PodTemplate from DaemonSet
-   `.spec.template`:
+   `.spec.revisionHistoryLimit` latest PodTemplates (remove the rest).
+  - Find the PodTemplate whose "pod-template-generation" label value equals to
+    DaemonSet's `.templateGeneration`. If not found, find the PodTemplate whose
+    "pod-template-hash" label value equals to the hash value of DaemonSet's
```

Comment:
Why do we need pod-template-hash? I think of this proposal as a better alternative to hashing and since both new labels need to be added we are ok with one of them (specifically the one about generation).